modules/preview-react/mutliselect/MultiSelectInput.tsx

```diff
@@ -0,0 +1,306 @@
+      //   console.log('click', model.state.visibility);
+      //   if (model.state.visibility === 'hidden') {
+      //     model.events.show(e);
+      //   } else if (model.state.visibility === 'visible') {
+      //     model.events.hide(e);
+      //   }
+      // },
+    };
+  }),
+  useComboboxInputConstrained,
+  useComboboxInput
+);
+
+const removeItem = <T extends unknown>(id: string, model: ReturnType<typeof useListModel>) => {
+  const index = model.state.items.findIndex(item => item.id === model.state.cursorId);
+  const nextIndex = index === model.state.items.length - 1 ? index - 1 : index + 1;
+  const nextId = model.state.items[nextIndex].id;
+  console.log('nextId', id, nextId);
+  if (model.state.cursorId === id) {
+    // We're removing the currently focused item. Focus next item
+    model.events.goTo({id: nextId});
+
+    // // wait for stabilization of state
+    // requestAnimationFrame(() => {
+    //   document.querySelector<HTMLElement>(`#${model.state.id}-${nextId}`)?.focus();
+    // });
+  }
+};
+
+const useMultiSelectedItem = composeHooks(
+  createElemPropsHook(useListModel)((model, ref, elemProps) => {
+    return {
+      onKeyDown(event: React.KeyboardEvent<HTMLElement>) {
+        const id = event.currentTarget.dataset.id || '';
+        if (event.key === 'Backspace' || event.key === 'Delete') {
+          model.events.select({id});
+          removeItem(id, model);
+        }
+      },
+      onClick(event: React.MouseEvent<HTMLElement>) {
+        const id = event.currentTarget.dataset.id || '';
+        model.events.select({id});
+      },
+    };
+  }),
+  useListItemRovingFocus,
+  useListItemRegister
+);
+
+const MultiSelectedItem = createSubcomponent('span')({
+  modelHook: useListModel,
+  elemPropsHook: useMultiSelectedItem,
+})(({children, ref, ...elemProps}, Element, model) => {
+  return (
+    <Pill as={Element} variant="removable">
+      {children}
+      <Pill.IconButton ref={ref} {...(elemProps as any)} />
+    </Pill>
+  );
+});
+
+export interface MultiSelectInputProps
+  extends CSProps,
+    Pick<
+      React.InputHTMLAttributes<HTMLInputElement>,
+      'disabled' | 'className' | 'style' | 'aria-labelledby'
+    > {}
+
+export const MultiSelectInput = createSubcomponent(TextInput)({
+  displayName: 'MultiSelect.Input',
+  modelHook: useMultiSelectModel,
+  elemPropsHook: useMultiSelectInput,
+})<MultiSelectInputProps>(
+  (
+    {className, cs, style, 'aria-labelledby': ariaLabelledBy, formInputProps, ...elemProps},
+    Element,
+    model
+  ) => {
+    return (
+      <div {...handleCsProp({className, cs, style}, multiSelectStencil({}))}>
+        <InputGroup>
+          <InputGroup.Input data-part="form-input" {...formInputProps} />
+          <InputGroup.Input
+            data-part="user-input"
+            as={Element}
+            aria-labelledby={ariaLabelledBy}
+            readOnly
+            {...elemProps}
+          />
+          <InputGroup.InnerEnd pointerEvents="none">
```

Comment:
```suggestion
          <InputGroup.InnerEnd cs={{pointerEvents: "none"}}>
```

Reply:
The `InputGroup.Inner*` have a `pointerEvents` variable/modifier combination. It is very common to change the pointer events of these inner items since icons can often be only for decoration.